core/src/drop_bank_service.rs

```diff
@@ -1,26 +1,27 @@
 use {
     crossbeam_channel::Receiver,
     solana_measure::measure::Measure,
-    solana_runtime::bank::Bank,
-    std::{
-        sync::Arc,
-        thread::{self, Builder, JoinHandle},
-    },
+    solana_runtime::installed_scheduler_pool::BankWithScheduler,
+    std::thread::{self, Builder, JoinHandle},
 };
 
 pub struct DropBankService {
     thread_hdl: JoinHandle<()>,
 }
 
 impl DropBankService {
-    pub fn new(bank_receiver: Receiver<Vec<Arc<Bank>>>) -> Self {
+    pub fn new(bank_receiver: Receiver<Vec<BankWithScheduler>>) -> Self {
         let thread_hdl = Builder::new()
             .name("solDropBankSrvc".to_string())
             .spawn(move || {
                 for banks in bank_receiver.iter() {
                     let len = banks.len();
                     let mut dropped_banks_time = Measure::start("drop_banks");
-                    drop(banks);
+                    // Drop BankWithScheduler with no alive lock to avoid deadlocks. That's because
+                    // BankWithScheduler::drop() could block on transaction execution if unified
+                    // scheduler is installed. As a historical context, it's dropped early inside
+                    // the replaying stage not here and that caused a deadlock for BankForks.
+                    drop::<Vec<BankWithScheduler>>(banks);
```

Comment:
seems odd to specify the type here to me, why?

Reply:
reason was that it's better off explicitly spelling the type mentioned by the preceding comment. guess this was a weak argument... just rmeoved: 8ef41a2c232a3871a6a7d3e819115e7941a61e62